chatty-cathy-client/src/main/java/org/client/Client.java

```diff
@@ -0,0 +1,70 @@
+package org.client;
+
+import org.common.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.net.*;
+
+public class Client {
+
+    private static final Logger logger = LoggerFactory.getLogger(Client.class);
+    private ObjectInputStream serverIn = null;
+
+    public Client(String addr, int port) throws IOException {
+        BufferedReader in;
+        ObjectOutputStream out;
+
+        Socket s;
+        try {
+            s = new Socket(addr, port);
+            logger.info("Connected");
+            in = new BufferedReader(new InputStreamReader(System.in));
+            out = new ObjectOutputStream(s.getOutputStream());
+            serverIn = new ObjectInputStream(s.getInputStream());
+
+            Thread readThread = new Thread(() -> {
+                try {
+                    while (true) {
+                        Message message = (Message) serverIn.readObject();
+                        logger.info(message.getUsername() + ": " + message.getMessage());
+                    }
+                } catch (IOException | ClassNotFoundException e) {
+                    logger.error(String.valueOf(e));
+                }
+            });
+            readThread.start();
+        } catch (IOException i) {
+            logger.error(String.valueOf(i));
+            return;
+        }
```

Comment:
A more conventional approach to logging the error **message** would be:
```java
        } catch (IOException e) {
            logger.error(e.getMessage());
            return;
        }
```
or `logger.error(e.toString());`

However, it is is best not to return just the message as it gets rid of the stack trace making it more difficult to find the source of issues.

To get the stack trace AND message simply log the error:
```suggestion
        } catch (IOException e) {
            logger.error(e);
            return;
        }
```

If you want to continue with just the message use that going forward for the next comments on this issue